predict-daily-temperature/Temperature-Demo.ipynb

```diff
@@ -0,0 +1,6470 @@
+    }
+   ],
+   "source": [
+    "training_es = ft.EntitySet()\n",
+    "training_es.add_dataframe(training_data.copy(), \n",
+    "                 dataframe_name='temperatures', \n",
+    "                 index='id', \n",
+    "                 make_index=True, \n",
+    "                 time_index=time_index)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "0eb6c9e3",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Entityset: None\n",
+       "  DataFrames:\n",
+       "    temperatures [Rows: 290, Columns: 3]\n",
+       "  Relationships:\n",
+       "    No relationships"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "test_es = ft.EntitySet()\n",
+    "test_es.add_dataframe(test_data.copy(), \n",
+    "                 dataframe_name='temperatures', \n",
+    "                 index='id', \n",
+    "                 make_index=True, \n",
+    "                 time_index=time_index)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "692e5759",
+   "metadata": {},
+   "source": [
+    "Now, we'll set up our primitives for DFS. There will be three types of primitives:\n",
+    "\n",
+    "- Datetime primitives - these extract features from datetime columns\n",
+    "- Rolling primitives - these perform aggregate calculations over rolling windows. Parameters can be set that allow an observation's window to be entirely before the observation occurs, which is important for not exposing our target variable. \n",
+    "- Lagging primitives - these are primitive versions of the delayed target feature from the baseline run. We'll define multiple of these with different delays.\n",
+    "\n",
+    "We use the `gap`, `forecast_horizon`, and `max_delay` parameters from before to set our primitive parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "id": "956d3b27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Rolling gap is comprised of both our `gap` parameter and the forecast horizon, \n",
+    "# because they determine the first observation we have access to. We'll also use them in defining our Numeric Lag primitives\n",
+    "rolling_gap = forecast_horizon + gap\n",
+    "\n",
+    "# The .25 is a heuristic to only use a portion of the max_delay for the window length\n",
+    "rolling_window_length = int(.25*max_delay) + 1 # a quarter is a heuristic here \n",
+    "# Min periods is a pandas parameter, and it just stops us from including partial calculations before the windows have \n",
+    "# all the possible observations\n",
+    "rolling_min_periods = int(.25*max_delay) + 1\n",
+    "\n",
+    "datetime_primitives = [ 'Month', 'Hour', \"Year\"]\n",
+    "delaying_primitives = [NumericLag(periods=t + forecast_horizon + gap) for t in range(forecast_horizon + gap + 1)]\n",
```

Comment:
I think the meaning of `max_delay` is more clearly explained if its `range(max_delay)`?

Reply:
Could it make sense to do `(.25*max_delay)` since that's what ends up getting set as the window length for the rolling primitives?

Reply:
Yea I think that makes sense!